core/src/sql/value/value.rs

```diff
@@ -2671,6 +2671,46 @@ impl Value {
 		}
 	}
 
+	/// Partially process this type (just the params) returning an evaluated simple Value that still needs computing
+	///
+	/// Is used recursively.
+	pub(crate) async fn partially_compute_unbordered(
+		&self,
+		stk: &mut Stk,
+		ctx: &Context<'_>,
+		opt: &Options,
+		doc: Option<&CursorDoc<'_>>,
+	) -> Result<Value, Error> {
+		// Prevent infinite recursion due to casting, expressions, etc.
+		let opt = &opt.dive(1)?;
+
+		match self {
+			// We only want to process params
+			Value::Param(v) => stk.run(|stk| v.compute(stk, ctx, opt, doc)).await,
```

Comment:
Do we want to exclude certain parameters? For example when a parameter is a `DEFINE PARAM`, it can change in the meantime, or `$auth` / `$this` / `$session` / `$scope` / `$access` / `$token` / other reserved variable names from https://surrealdb.com/docs/surrealdb/surrealql/parameters#reserved-variable-names

Reply:
Oof yeah that needs to change now, thanks

Reply:
Resolved